src/NUnitFramework/tests/Assertions/AssertFailTests.cs

```diff
@@ -104,13 +104,13 @@ public void AssertFailWorksWithMessageAndArgs()
             Assert.AreEqual(ResultState.Failure, result.ResultState);
             Assert.AreEqual("MESSAGE: 2+2=4", result.Message);
 
-            Assert.AreEqual(result.AssertionResults.Count, 1);
+            Assert.AreEqual(1, result.AssertionResults.Count);
             var assertion = result.AssertionResults[0];
             Assert.That(assertion.Status, Is.EqualTo(AssertionStatus.Failed));
             Assert.That(assertion.Message, Is.EqualTo("MESSAGE: 2+2=4"));
         }
 
-        [Test]
+        [Test, Ignore("Currently Fails: Must use Assert.Catch or Assert.Throws")]
```

Comment:
As mentioned in the issue, I'm personally not fond of this breaking change. Even though NUnit has this goal of separating test termination from the failures, it just results in unintuitive behavior to the end user.

Reply:
Agreed. OTOH, being able to catch an exception thereby preventing a failure is purely an accident of implementation, even though it's an "obvious" implementation. Note that you have __never__ been above to prevent a multiple assert failure by catching an exception and - of course - you can't prevent warnings that way either.

The only alternative I can see is that asserts not use the current result at all, but just make and update a copy of it, which they then throw. That seems like quite a lot of work and I'd want to apply it to all types of exceptions uniformly if we did it.

Reply:
@julealgon Which of the two approaches were you giving the thumbs up to?

Reply:
> @julealgon Which of the two approaches were you giving the thumbs up to?

I was just agreeing with your comment in general, in that I understand the huge challenge it would be to do something different right now, even though I'm still not fond of the proposed approach.

As I said, it is infinitely more intuitive to control failures _and_  test termination via the same exceptions, but it looks like the entire team is set on the idea of having these concerns separated. I'm leaving this one up to you guys.